### PR TITLE
[2304] Course dates for routes which should refer to itt dates for itt trainees

### PR DESCRIPTION
**Pull request by defong**
### Context

ITT vs non ITT start and end dates

### Changes proposed in this pull request
`Teaching apprenticeship (postgrad)` route, everything should say `ITT start date` or `ITT end date`

### Guidance to review
Go thro the `Teaching apprenticeship (postgrad)` route, everything should say `ITT start date` or `ITT end date`

https://register-pr-1195.london.cloudapps.digital/trainees/SSLModxSnzXHQKSSTMVdMnuJ

image

https://register-pr-1195.london.cloudapps.digital/trainees/A3NqSjdZ34bzS8bb5Y5TmFRk/course-details/edit

image

https://register-pr-1195.london.cloudapps.digital/trainees/A3NqSjdZ34bzS8bb5Y5TmFRk/check-details
image

https://register-pr-1195.london.cloudapps.digital/trainees/A3NqSjdZ34bzS8bb5Y5TmFRk/course-details/confirm

image

This should provisional work for the other routes 
- `early_years_undergrad`, 
- `provider_led_undergrad`, 
- `opt_in_undergrad`

### CODE REVIEW

**Comment by defong** [comment]
[Update form and summary labels for course dates for routes which should refer to ITT dates](https://trello.com/c/zd4KmIpR/2304-update-form-and-summary-labels-for-course-dates-for-routes-which-should-refer-to-itt-dates)

File: spec/lib/training_route_manager_spec.rb
Comment (gpeng) [inline spec/lib/training_route_manager_spec.rb:76]: I find these really hard to follow. Might just be me obvs but I prefer the duplication in specs as it makes them easier to read. Doesn't need to block this PR but we should see what the rest of the team think and add whatever we decide to the testing guidelines doc we have.

**Review by gpeng — COMMENTED** [review]
Do we need to change the field name on the form if we're not persisting it to a different DB field? It seems to add a lot of complexity vs just changing the text on the page. Also could we squash the 'fix typo' commit into wherever you 'did typo' 🤣

**Comment by defong** [comment]
> Also could we squash the 'fix typo' commit into wherever you 'did typo' rofl

squashed the 'fix typo'

File: spec/lib/training_route_manager_spec.rb
Comment (gpeng) [reply spec/lib/training_route_manager_spec.rb:76]: We discussed this offline and the team decided that in general we should avoid shared examples and test explicitly. Not an absolute rule if shared examples make something easier to understand though.

Comment (defong) [reply spec/lib/training_route_manager_spec.rb]: redone as https://github.com/DFE-Digital/register-trainee-teachers/pull/1195/commits/0f477c0f51d992a544ca52497bdb9b6ba75b6cf4

**Review by gpeng — APPROVED** [review]
Code looks ok. I think the error messages should be 'Enter an ITT' rather than 'Enter a'. Also not sure about the 'as far in the future' bit but I'll defer to @gregknight1 for that. Need to check with him before we merge.

**Comment by defong** [comment]
image

Updated as discussed with @gregknight1

**Comment by gregknight1** [comment]
Looks good @defong

**Comment by bbelward** [comment]
🤔 not seeing ITT start date on the confirmation page and getting an error - am in the Course details section
ITTstartdate-error

**Comment by defong** [comment]
image
Also fixed poor validation html 

Thanks @edwardhorsford

**Pull request merged**
